src/vs/gitpod/browser/gitpodInsightsAppender.ts

```diff
@@ -22,15 +25,87 @@ export class GitpodInsightsAppender implements ITelemetryAppender {
 	}
 
 	public log(eventName: string, data: any): void {
-		const trackMessage = mapTelemetryData(SenderKind.Browser, eventName, data);
-		if (!trackMessage) {
-			return;
+		this.sendAnalytics(eventName, data);
+		this.sendMetrics(eventName, data);
+	}
+
+	private sendAnalytics(eventName: string, data: any): void {
+		try {
+			const trackMessage = mapTelemetryData('window', eventName, data);
+			if (!trackMessage) {
+				return;
+			}
+			trackMessage.properties = {
+				...trackMessage.properties,
+				...this._baseProperties,
+			};
+			if (this.devMode) {
+				if (this.productService.gitpodPreview?.log?.analytics) {
+					console.log('Gitpod Analytics: ', JSON.stringify(trackMessage, undefined, 2));
+				}
+			} else {
+				// TODO(ak) get rid of it
+				// it is bad usage of window.postMessage
+				// we should use Segment directly here and publish to production/staging untrusted
+				// use server api to resolve a user
+				window.postMessage({ type: 'vscode_telemetry', event: trackMessage.event, properties: trackMessage.properties }, '*');
+			}
+		} catch (e) {
+			console.error('failed to send IDE analytic:', e);
 		}
-		trackMessage.properties = {
-			...trackMessage.properties,
-			...this._baseProperties,
-		};
-		window.postMessage({ type: 'vscode_telemetry', event: trackMessage.event, properties: trackMessage.properties }, '*');
+	}
+
+	private async sendMetrics(eventName: string, data: any): Promise<void> {
```

Comment:
The browser gitpodInsightsAppender.ts is just and edge for the reconnection events when we lose connection to the remote server, all other events will be send to the remote so not sure if we need the `sendMetrics` logic here